src/main.py

```
@@ -0,0 +1,105 @@
+        return json.loads(json_string, object_hook=JsonFile.json_decoder)
+
+
+class ReleaseVersion:
+    def __init__(self, file_name, file_content, version):
+        self.file_name = file_name
+        self.file_content = file_content
+        self.version = version
+
+    @staticmethod
+    def last_released_version() -> str:
+        """
+        Returns last released version from GitHub tag
+        :return: str
+        """
+        repository_name = sys.argv[1]
+        repository_to_clone = Definition.SHOP_EXTENSION_PARTNER + "/" + repository_name
+        return lastversion.latest(repository_to_clone, output_format='version', pre_ok=True)
+
+    @staticmethod
+    def current_release_version() -> str:
+        """
+        Returns current release version from branch name
+        :return: str
+        """
+        repo = git.Repo(search_parent_directories=True)
+        branch = repo.active_branch
+        return re.sub('[^\d\.]', '', branch.name)
+
+    def version_differences(self) -> list:
+        """
+        Returns list of version differences
+        :return: list
+        """
+        self.file_name = open(path.abspath(subprocess.check_output("find . -name " + str(self.file_name), shell=True, text=True)).rstrip(), 'r')
+        for file_line in self.file_name.readlines():
+            if self.version in file_line and str(ReleaseVersion.last_released_version()) in file_line:
+                self.file_content.append(file_line.replace(str(ReleaseVersion.last_released_version()), ReleaseVersion.current_release_version()))
+            else:
+                self.file_content.append(file_line)
+        return self.file_content
+
+
+class ExtensionVersionUpdater:
+    def __init__(self, extension_name):
+        self.extension_name = extension_name
+
+    def update_release_version(self):
+        naming_convention = Definition.EXTENSION_NAMING_CONVENTION
+        if self.extension_name in naming_convention:
```

maybe we could refactor here to fail fast - if self.extension_name not  in naming_convention - throw some exception